source/sass/views/youtube-regrets.scss

```diff
@@ -3,6 +3,170 @@
+
+    $numRings: 5;
+
+    min-height: 100vh;
+    height: calc(var(--sceneDepth) * 1px);
+
+    .scene-container {
+      position: fixed;
+      top: 0;
+      left: 0;
+      width: 100%;
+      height: 100%;
+
+      .scene {
+        position: absolute;
+        left: 0;
+        right: 0;
+        top: 0;
+        bottom: 0;
+        margin: auto;
+        transform-style: preserve-3d;
+        transform: translate3D(0, 0, 0);
+        will-change: transform;
+        perspective-origin: 50% 50%;
+        perspective: calc(var(--scenePerspective) * 1px);
+        width: 100vw;
+        height: 100vw;
+
+        @media (min-width: $bp-sm) {
+          width: 60vw;
+          height: 60vw;
+        }
+
+        .block {
+          position: absolute;
+          display: block;
+          width: 100%;
+          opacity: 0;
+
+          @function get-block-z-tranlsate($block-index) {
+            @return calc(
+              var(--blockZTranslate) * 1px -
+                calc(var(--baseBlockGap) * #{$block-index}) * 1px +
+                var(--scenePerspective) * 1px
+            );
+          }
+
+          &:nth-child(1) {
+            transform: translate3D(-100%, 250%, get-block-z-tranlsate(1));
+          }
+
+          &:nth-child(2) {
+            transform: translate3D(240%, 240%, get-block-z-tranlsate(2));
+          }
+
+          &:nth-child(3) {
+            transform: translate3D(-200%, -290%, get-block-z-tranlsate(3));
+          }
+
+          &:nth-child(4) {
+            transform: translate3D(270%, -230%, get-block-z-tranlsate(4));
+          }
+
+          &:nth-child(5) {
+            transform: translate3D(50%, 150%, get-block-z-tranlsate(5));
+          }
+
+          &:nth-child(6) {
+            transform: translate3D(-220%, -180%, get-block-z-tranlsate(6));
+          }
+
+          &:nth-child(7) {
+            transform: translate3D(300%, -250%, get-block-z-tranlsate(7));
+          }
+
+          &:nth-child(8) {
+            transform: translate3D(-50%, 170%, get-block-z-tranlsate(8));
+          }
+
+          &:nth-child(9) {
+            transform: translate3D(130%, -220%, get-block-z-tranlsate(9));
+          }
+
+          &:nth-child(10) {
+            transform: translate3D(60%, 250%, get-block-z-tranlsate(10));
+          }
+        }
+
+        .ring {
+          position: absolute;
+          border-radius: 50%;
+          opacity: 0;
+          width: 100%;
+          height: 100%;
+          box-shadow: inset 0 0 5px 5px $white;
+
+          @media (min-width: $bp-sm) {
+            box-shadow: inset 0 0 10px 15px $white;
+          }
+
+          @function get-ring-z-tranlsate($ring-index) {
+            @return calc(
+              1px *
+                (
+                  var(--ringZTranslate) +
+                  var(--baseRingGap) * #{$ring-index} -
+                  var(--scenePerspective) / 2
+                )
+            );
+          }
+
+          @for $i from 0 through $numRings {
```

Comment:
What does this bit do? With five rings, it feels like this will get run 25 times?

Reply:
it's just like a for loop run from index 0 to 4